sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -9,14 +9,14 @@
 #include "azure/core/nullable.hpp"
 #include "azure/core/response.hpp"
 #include "azure/storage/common/crypt.hpp"
-#include "azure/storage/common/json.hpp"
 #include "azure/storage/common/storage_common.hpp"
 #include "azure/storage/common/storage_exception.hpp"
 
 #include <functional>
 #include <iostream>
 #include <map>
 #include <memory>
+#include <nlohmann/json.hpp>
```

Comment:
We need to think carefully before taking a dependency and publicly exposing a third party header like this.

Reply:
there's possible workaround

```
namspace Details
{
#include <nlohmann/json.hpp>
}
```

Reply:
Yeah, I think this is actually a bug. Headers shouldn't be including this, only cpp files. 
The problem is that this header contains implementation. 
Instead of the Details workaround, Storage team should move the implementation to cpp files and let the header just for the signatures. @Jinming-Hu 

That fix can be done in a different PR since it is not currently the main intention of the actual PR.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/1009